<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<!-- Real Turtlebot -->
        <include file="$(find turtlebot_bringup)/launch/minimal.launch" />

	<!-- Some general parameters -->
	<rosparam file="$(find nav2d_tutorials)/param/ros.yaml"/>


	<!-- Start the Operator to control the simulated robot -->
	<node name="Operator" pkg="nav2d_operator" type="operator" >
		<remap from="scan" to="base_scan"/>
		<rosparam file="$(find nav2d_tutorials)/param/operator.yaml"/>
		<rosparam file="$(find nav2d_tutorials)/param/costmap.yaml" ns="local_map" />
	</node>

	<!-- Start Mapper to generate map from laser scans -->
	<node name="Mapper" pkg="nav2d_karto" type="mapper">
		<remap from="cmd_vel" to="cmd_vel_mux/input/navi"/>
		<remap from="scan" to="base_scan"/>
		<rosparam file="$(find nav2d_tutorials)/param/mapper.yaml"/>
	</node>

	<!-- Start the Navigator to move the robot autonomously -->
	<node name="Navigator" pkg="nav2d_navigator" type="navigator">
		<rosparam file="$(find nav2d_tutorials)/param/navigator.yaml"/>
	</node>

	<node name="GetMap" pkg="nav2d_navigator" type="get_map_client" />
	<node name="Explore" pkg="nav2d_navigator" type="explore_client" />
	<node name="SetGoal" pkg="nav2d_navigator" type="set_goal_client" />

</launch>
